aws/resource_aws_lambda_layer.go

```diff
@@ -0,0 +1,262 @@
+	}
+
+	log.Printf("[DEBUG] Publishing Lambda layer: %s", params)
+	result, err := conn.PublishLayerVersion(params)
+	if err != nil {
+		return fmt.Errorf("Error creating lambda layer: %s", err)
+	}
+
+	d.SetId(aws.StringValue(result.LayerVersionArn))
+	return resourceAwsLambdaLayerRead(d, meta)
+}
+
+func resourceAwsLambdaLayerRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).lambdaconn
+
+	layerName, version, err := resourceAwsLambdaLayerParseId(d.Id())
+	if err != nil {
+		return fmt.Errorf("Error parsing lambda layer ID: %s", err)
+	}
+
+	layerVersion, err := conn.GetLayerVersion(&lambda.GetLayerVersionInput{
+		LayerName:     aws.String(layerName),
+		VersionNumber: aws.Int64(version),
+	})
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "ResourceNotFoundException" && !d.IsNewResource() {
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	if err := d.Set("layer_name", layerName); err != nil {
+		return fmt.Errorf("Error setting lambda layer name: %s", err)
+	}
+	if err := d.Set("version", strconv.FormatInt(version, 10)); err != nil {
+		return fmt.Errorf("Error setting lambda layer version: %s", err)
+	}
+	if err := d.Set("arn", layerVersion.LayerArn); err != nil {
+		return fmt.Errorf("Error setting lambda layer arn: %s", err)
+	}
+	if err := d.Set("qualified_arn", layerVersion.LayerVersionArn); err != nil {
+		return fmt.Errorf("Error setting lambda layer qualified arn: %s", err)
+	}
+	if err := d.Set("description", layerVersion.Description); err != nil {
+		return fmt.Errorf("Error setting lambda layer description: %s", err)
+	}
+	if err := d.Set("license_info", layerVersion.LicenseInfo); err != nil {
+		return fmt.Errorf("Error setting lambda layer license info: %s", err)
+	}
+	if err := d.Set("created_date", layerVersion.CreatedDate); err != nil {
+		return fmt.Errorf("Error setting lambda layer created date: %s", err)
+	}
+	if err := d.Set("source_code_hash", layerVersion.Content.CodeSha256); err != nil {
+		return fmt.Errorf("Error setting lambda layer source code hash: %s", err)
+	}
+	if err := d.Set("source_code_size", layerVersion.Content.CodeSize); err != nil {
+		return fmt.Errorf("Error setting lambda layer source code size: %s", err)
+	}
+	if err := d.Set("compatible_runtimes", flattenStringList(layerVersion.CompatibleRuntimes)); err != nil {
+		return fmt.Errorf("Error setting lambda layer compatible runtimes: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsLambdaLayerDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).lambdaconn
+
+	version, err := strconv.ParseInt(d.Get("version").(string), 10, 64)
+	if err != nil {
+		return fmt.Errorf("Error parsing lambda layer version: %s", err)
+	}
+
+	_, err = conn.DeleteLayerVersion(&lambda.DeleteLayerVersionInput{
+		LayerName:     aws.String(d.Get("layer_name").(string)),
+		VersionNumber: aws.Int64(version),
+	})
+	if err != nil {
+		return err
```

Comment:
Nit: We should provide context about when this error occurred for operators and code maintainers, e.g.

```suggestion
		return fmt.Errorf("error deleting Lambda Layer Version (%s): %s", d.Id(), err)
```